### PR TITLE
Migrate CI to GitHub Actions 

### PR DESCRIPTION
**Pull request by petergoldstein**
This PR migrates CI to GitHub actions.  Linting is split out into a separate task from specs.  Everything is green.

As pry-byebug is not compatible with Ruby 3.2 and beyond, it's been removed from the project so that ruby-head runs green.

An additional Code Climate id is required to get Code Climate working with the new setup.

### CODE REVIEW

**Review by waiting-for-dev — APPROVED** [review]
Thanks, @petergoldstein, that's very nice!

I added one comment, but I can do it myself in another commit. Also, it'd be nice to remove `.travis.yml` as part of this PR.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: CI 
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby-version: ['2.6', '2.7', '3.0', '3.1', ruby-head]
```
[the remark this answers is not in the document]
Comment (petergoldstein) [reply .github/workflows/ci.yml]: Done.

**Comment by petergoldstein** [comment]
@waiting-for-dev I've removed .travis.yml and Ruby 2.6.

**Comment by waiting-for-dev** [comment]
Thanks! :heart:

**Pull request merged**